src/Plugin/Field/FieldWidget/EthereumAddressWidget.php

```diff
@@ -35,6 +35,10 @@ public static function defaultSettings() {
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
     $element = parent::formElement($items, $delta, $element, $form, $form_state);
 
+    if ($element['value']['#default_value']) {
+      $element['value']['#attributes']['data-original-value'] = $element['value']['#default_value'];
```

Comment:
Why do we need the original value here?

Reply:
I use it to check if a address entry is the same than before, in order to give the address_status the right value. You are right: technically it should be somewhere in ethereum_user_connector module. There are 3 fields on the user required to manage the contract registry subscription (address, hash, address_status).